### ISSUE TITLE
False positive on AvoidImplicitlyRecompilingPatterns for replaceAll

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#4]
If the first argument to String replaceAll isn't actually a regex, it still reports a violation.
Checking String.split is fixed for this case. Checking other methods should be fixed as well, where applicable.

Test case from project code:
```
    public final static String PAYLOADDATA = "PAYLOADDATA";
    private String stringReplaceAllNonRegexOK() {
        String result = "some xml data with [PAYLOADDATA]";
        StringBuilder payloadData = new StringBuilder("some payload");
        result = result.replaceAll(PAYLOADDATA, payloadData.toString());
        return result;
    }
```

**Comment by stokpop** [comment]
created fix in this pull request: https://github.com/jborgers/PMD-jPinpoint-rules/pull/11

**Comment by jborgers** [comment]
Fixed
